src/App.tsx

```diff
@@ -1,39 +1,56 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+type State = {
+  showClock: boolean
+  clockName: string
+};
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    showClock: true,
+    clockName: 'Clock-0',
+  };
 
-        {' time is '}
+  timerId = window.setInterval(() => {
+    this.setState({ clockName: getRandomName() });
+  }, 3300);
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+
+      this.setState({ showClock: false });
+    });
+
+    document.addEventListener('click', () => {
+      this.setState({ showClock: true });
+    });
+  }
+
+  componentDidUpdate(prevProps:State) {
+    console.log(
```

Comment:
you have lint error here, change to debug

Reply:
i added eslint-disable-next-line no-console, if it is correct, if no what exelent i should add?

Reply:
console.debug